step/src/main/java/com/google/sps/servlets/PrefServlet.java

```diff
@@ -49,6 +69,11 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
             .setFilter(new Query.FilterPredicate("id", Query.FilterOperator.EQUAL, id));
     PreparedQuery results = datastore.prepare(query);
     Entity entity = results.asSingleEntity();
+
+    if (entity == null) {
+      response.sendRedirect("index.jsp");
+    }
```

Comment:
Can you add a pop up for the user, something like "Your profile couldn't be loaded"?

Reply:
Going to add this one to the spreadsheet because it might take a bit and there's other tasks I want to get to first.